docs/openstack-cloud-design-overview.md

```diff
@@ -0,0 +1,52 @@
+# Overview
+
+When building a cloud, the design of the cloud will not only reflect the type of services you are looking to provide but also inform the way that users of your cloud work to accomplish their goals.
+
+Availability is one of the key aspects of deploying applications in the cloud.  Whether you are building a [twelve-factor](https://12factor.net/){:target="_blank"} cloud-native application or running a more traditional stateful application,  in order for an application to maximize availability, it needs to have resiliance against failures, and that means understanding what failure domains the cloud infrastructure has.
+
+As the cloud designer and builder, providing the capabilities for high-availability means effectively defining your cloud structure for users to work around these failure domains.  Depending on the resources you have or wish to make availabile to your users, you should decide how to set up the topology and architecture of your cloud to enable users to plan their deployments to maximize the availabilty of their applications.  This clould topology consists of the hierarchy of organizational units of your cloud and decide how your services function across them.  Examples of these organizational units include:
+
+- Clouds
+- Regions
+- Availability Zones
+
+These are typically arranged as in a hierarchy, with each incresing lower layer becoming less independent of the other layers:
+
+![Cloud Hierarchy](assets/images/cloud-hierarchy.png)
+
+## Clouds
+
+At the highest level, a cloud is a  unique set of services that  operates completely independedntly.  We can define a cloud as having its own set of resources and services that are not dependent on any other cloud.
```

Comment:
A couple of places with double spaces but that's fine, I know it's a WIP, just making sure you know. LGTM